libcxx/include/__algorithm/ranges_stable_sort.h

```diff
@@ -45,7 +47,19 @@ struct __stable_sort {
     auto __last_iter = ranges::next(__first, __last);
 
     auto&& __projected_comp = std::__make_projected(__comp, __proj);
-    std::__stable_sort_impl<_RangeAlgPolicy>(std::move(__first), __last_iter, __projected_comp);
+    constexpr auto __default_comp =
+        __desugars_to_v<__totally_ordered_less_tag, _Comp, iter_value_t<_Iter>, iter_value_t<_Iter> >;
+    constexpr auto __default_proj        = __is_identity<_Proj>::value;
+    constexpr auto __integral_value      = is_integral_v<iter_value_t<_Iter>>;
+    constexpr auto __integral_projection = __default_proj && __integral_value;
```

Comment:
If we do the dispatching in `__stable_sort_impl` instead we don't need to do the dispatching twice.

Reply:
Done.